### DISCUSSION TITLE
Regarding Collection level pre scripts

### DISCUSSION THREAD

**Discussion by altereddie** [thunderclient/thunder-client-support#1401] [General]
So i am trying to replicate our requirement into thunder client from postman. 

So we have script at collection level in postman, and we are using that pre script into api level pre script 

Attaching code for better understanding 

test(collection) > testx(api)

**POSTMAN**

test (collection) inside pre-request script section 

`utils = {
  myFunc: function () {
    return 'hello';
  }
};
`
and in testx(api) inside pre script section 

`console.log(utils.myFunc()); // hello`

and it works

now in **THUNDER CLIENT**

 test (collection) inside pre-run scripting section 
 
 `utils = {
  myFunc: function () {
    return 'hello';
  }
};
`
and in testx(api) inside pre scripting section 

`
console.log(utils.myFunc()); // hello
`

and this thing dont work. 

can you look into it and guide me ?

**Comment by rangav** [comment]
Hi @altereddie 


We have two ways of writing scripts
- [Inline Scripting](https://github.com/rangav/thunder-client-support/blob/master/docs/filters.md#inline-scripting) 
- [External JS files](https://github.com/rangav/thunder-client-support/blob/master/docs/custom-filters.md) 

If you need reusable functions then you can try External JS files above

You can also use both scripts features above.

Let me know your feedback.

**Comment by rangav** [comment]
### Import functions from other js files in Scripts

Just now implemented this feature - Now you can import as below in the next update **(ETA - Monday)**

```js
async function testFunc () {
    console.log ("Test log message");
    return "TEST";
  }

  module.exports = [testFunc];
```


in this collection there is one testx api inside that i put tests script as following 

```js
const [testFunc] = require("test.js");

const result = await testFunc();
console.log(result); // TEST
console.log('test hello')
```

**Comment by altereddie** [comment]
@rangav 
Hey pal. Thanks for the quick support and response. :)

**Comment by rangav** [comment]
This feature is implemented and published to the marketplace, please update to `v2.16.0`

See all features in this update
https://github.com/rangav/thunder-client-support/releases/tag/v2.16.0

Please test and let me know your feedback.
